File: budgetmap-api/src/main/resources/static/fix_tildes.js
const fs = require('fs');
const path = require('path');

const dir = "c:/PROYECTO REAL/budgetmap-api/src/main/resources/static";

function walk(dir) {
    let results = [];
    const list = fs.readdirSync(dir);
    list.forEach(function(file) {
        file = dir + '/' + file;
        const stat = fs.statSync(file);
        if (stat && stat.isDirectory()) { 
            results = results.concat(walk(file));
        } else { 
            if (file.endsWith('.html')) results.push(file);
        }
    });
    return results;
}

const htmlFiles = walk(dir);

const replacements = {
    "Descripcin": "Descripción",
    "Trminos": "Términos",
    "Telfono": "Teléfono",
    "Mximo": "Máximo",
    "Ubicacin": "Ubicación",
    "Direccin": "Dirección",
    "Gestin": "Gestión",
    "Informacin": "Información",
    "Configuracin": "Configuración",
    "Contrasea": "Contraseña",
    "Validacin": "Validación",
    "Prximos": "Próximos",
    "conexin": "conexión",
    "crtico": "crítico",
    "vlida": "válida",
    "CDIGO": "CÓDIGO",
    "Cdigo": "Código",
    "PsBLICO": "PÚBLICO",
    "Aadir": "Añadir",
    "Reserva": "¡Reserva",
    "Informacin": "¡Información",
    "Y\"?": "📍",
    "Y\".": "📅",
    "Y??": "🏁",
    "YZY?": "🎫",
    "Categora": "Categoría",
    "categora": "categoría",
    "da": "día",
    "Da": "Día",
    "Ao": "Año",
    "ao": "año",
    "Accin": "Acción",
    "accin": "acción",
    "seccin": "sección",
    "Seccin": "Sección",
    "Ttulo": "Título",
    "ttulo": "título",
    "Mǭs": "Más",
    "mǭs": "más",
    "Aqu": "Aquí",
    "aqu": "aquí",
    "Aadir": "Añadir",
    "aadir": "añadir",
    "Inici": "Inició",
    "inici": "inició",
    "TambiǸn": "También",
    "tambiǸn": "también",
    "Estǭ": "Está",
    "estǭ": "está",
    "S": "Sí",
    "s": "sí",
    "Opciones de Gestin": "Opciones de Gestión",
    "Previsualizacin": "Previsualización",
    "PsBLICO": "PÚBLICO",
    "C\"DIGO": "CÓDIGO",
    "Descripcin": "Descripción",
    "Trminos": "Términos",
    "Telfono": "Teléfono",
    "Mximo": "Máximo",
    "Ubicacin": "Ubicación",
    "Direccin": "Dirección",
    "Gestin": "Gestión",
    "Informacin": "Información",
    "Configuracin": "Configuración",
    "Contrasea": "Contraseña",
    "Validacin": "Validación",
    "Prximos": "Próximos",
    "conexin": "conexión",
    "crtico": "crítico",
    "vlida": "válida",
    "Cdigo": "Código",
    "Reserva": "¡Reserva",
    "Informacin": "¡Información",
    "Y\"?": "📍",
    "Y\".": "📅",
    "Y??": "🏁",
    "YZ": "🎫",
    "YZY": "🎫"
};

for (const f of htmlFiles) {
    let content = fs.readFileSync(f, 'utf8');
    
    // Quick replacements based on the object
    let changed = false;
    for (const [bad, good] of Object.entries(replacements)) {
        if (content.includes(bad)) {
            content = content.split(bad).join(good);
            changed = true;
        }
    }
    
    // Also use the actual unicode replacement character
    const corruptChar = '\ufffd';
    const genericFixes = {
        [`Descripci${corruptChar}n`]: "Descripción",
        [`Descripci${corruptChar}`]: "Descripción",
        [`T${corruptChar}rminos`]: "Términos",
        [`Tel${corruptChar}fono`]: "Teléfono",
        [`M${corruptChar}ximo`]: "Máximo",
        [`Ubicaci${corruptChar}n`]: "Ubicación",
        [`Direcci${corruptChar}n`]: "Dirección",
        [`Gesti${corruptChar}n`]: "Gestión",
        [`Informaci${corruptChar}n`]: "Información",
        [`Configuraci${corruptChar}n`]: "Configuración",
        [`Contrase${corruptChar}a`]: "Contraseña",
        [`Validaci${corruptChar}n`]: "Validación",
        [`Pr${corruptChar}ximos`]: "Próximos",
        [`conexi${corruptChar}n`]: "conexión",
        [`cr${corruptChar}tico`]: "crítico",
        [`v${corruptChar}lida`]: "válida",
        [`C${corruptChar}DIGO`]: "CÓDIGO",
        [`C${corruptChar}digo`]: "Código",
        [`P${corruptChar}sBLICO`]: "PÚBLICO",
        [`A${corruptChar}adir`]: "Añadir",
        [`${corruptChar}Reserva`]: "¡Reserva",
        [`${corruptChar}Informaci${corruptChar}n`]: "¡Información",
        [`C${corruptChar}"DIGO`]: "CÓDIGO",
        [`${corruptChar}Y"?`]: "📍",
        [`${corruptChar}Y".`]: "📅",
        [`${corruptChar}Y??`]: "🏁",
        [`${corruptChar}YZY?`]: "🎫",
        [`Categor${corruptChar}a`]: "Categoría",
        [`categor${corruptChar}a`]: "categoría",
        [`d${corruptChar}a`]: "día",
        [`D${corruptChar}a`]: "Día",
        [`A${corruptChar}o`]: "Año",
        [`a${corruptChar}o`]: "año",
        [`Acci${corruptChar}n`]: "Acción",
        [`acci${corruptChar}n`]: "acción",
        [`secci${corruptChar}n`]: "sección",
        [`Secci${corruptChar}n`]: "Sección",
        [`T${corruptChar}tulo`]: "Título",
        [`t${corruptChar}tulo`]: "título",
        [`M${corruptChar}s`]: "Más",
        [`m${corruptChar}s`]: "más",
        [`Aqu${corruptChar}`]: "Aquí",
        [`aqu${corruptChar}`]: "aquí",
        [`A${corruptChar}adir`]: "Añadir",
        [`a${corruptChar}adir`]: "añadir",
        [`Inici${corruptChar}`]: "Inició",
        [`inici${corruptChar}`]: "inició",
        [`Tambi${corruptChar}n`]: "También",
        [`tambi${corruptChar}n`]: "también",
        [`Est${corruptChar}`]: "Está",
        [`est${corruptChar}`]: "está",
        [`S${corruptChar}`]: "Sí",
        [`s${corruptChar}`]: "sí",
        [`Opciones de Gesti${corruptChar}n`]: "Opciones de Gestión",
        [`Previsualizaci${corruptChar}n`]: "Previsualización"
    };

    for (const [bad, good] of Object.entries(genericFixes)) {
        if (content.includes(bad)) {
            content = content.split(bad).join(good);
            changed = true;
        }
    }
    
    if (changed) {
        fs.writeFileSync(f, content, 'utf8');
        console.log("Fixed:", f);
    }
}
console.log("Done");

File: budgetmap-api/FixTildes.java
import java.nio.file.*;
import java.nio.charset.StandardCharsets;
import java.io.IOException;

public class FixTildes {
    public static void main(String[] args) throws IOException {
        Path start = Paths.get("src/main/resources/static");
        Files.walk(start)
             .filter(Files::isRegularFile)
             .filter(p -> p.toString().endsWith(".html"))
             .forEach(p -> {
                 try {
                     String content = new String(Files.readAllBytes(p), StandardCharsets.UTF_8);
                     boolean changed = false;
                     
                     String[] bad = { "Descripcin", "Trminos", "Telfono", "Mximo", "Ubicacin", "Direccin", "Gestin", "Informacin", "Configuracin", "Contrasea", "Validacin", "Prximos", "conexin", "crtico", "vlida", "CDIGO", "Cdigo", "PsBLICO", "Aadir", "Reserva", "Informacin", "Categora", "categora", "Ttulo", "ttulo", "Opciones de Gestin", "Previsualizacin" };
                     String[] good = { "Descripción", "Términos", "Teléfono", "Máximo", "Ubicación", "Dirección", "Gestión", "Información", "Configuración", "Contraseña", "Validación", "Próximos", "conexión", "crítico", "válida", "CÓDIGO", "Código", "PÚBLICO", "Añadir", "¡Reserva", "¡Información", "Categoría", "categoría", "Título", "título", "Opciones de Gestión", "Previsualización" };

                     for (int i=0; i<bad.length; i++) {
                         if (content.contains(bad[i])) {
                             content = content.replace(bad[i], good[i]);
                             changed = true;
                         }
                     }
                     
                     char c = '\uFFFD';
                     String[] badC = { "Descripci"+c+"n", "Descripci"+c, "T"+c+"rminos", "Tel"+c+"fono", "M"+c+"ximo", "Ubicaci"+c+"n", "Direcci"+c+"n", "Gesti"+c+"n", "Informaci"+c+"n", "Configuraci"+c+"n", "Contrase"+c+"a", "Validaci"+c+"n", "Pr"+c+"ximos", "conexi"+c+"n", "cr"+c+"tico", "v"+c+"lida", "C"+c+"DIGO", "C"+c+"digo", "P"+c+"sBLICO", "A"+c+"adir", c+"Reserva", c+"Informaci"+c+"n", "C"+c+"\"DIGO", "Categor"+c+"a", "categor"+c+"a", "Acci"+c+"n", "acci"+c+"n", "secci"+c+"n", "Secci"+c+"n", "T"+c+"tulo", "t"+c+"tulo", "Opciones de Gesti"+c+"n", "Previsualizaci"+c+"n" };
                     String[] goodC = { "Descripción", "Descripción", "Términos", "Teléfono", "Máximo", "Ubicación", "Dirección", "Gestión", "Información", "Configuración", "Contraseña", "Validación", "Próximos", "conexión", "crítico", "válida", "CÓDIGO", "Código", "PÚBLICO", "Añadir", "¡Reserva", "¡Información", "CÓDIGO", "Categoría", "categoría", "Acción", "acción", "sección", "Sección", "Título", "título", "Opciones de Gestión", "Previsualización" };

                     for (int i=0; i<badC.length; i++) {
                         if (content.contains(badC[i])) {
                             content = content.replace(badC[i], goodC[i]);
                             changed = true;
                         }
                     }

                     String[] badC2 = { c+"Y\"?", c+"Y\".", c+"Y??", c+"YZY?" };
                     String[] goodC2 = { "📍", "📅", "🏁", "🎫" };
                     for (int i=0; i<badC2.length; i++) {
                         if (content.contains(badC2[i])) {
                             content = content.replace(badC2[i], goodC2[i]);
                             changed = true;
                         }
                     }

                     if (changed) {
                         Files.write(p, content.getBytes(StandardCharsets.UTF_8));
                         System.out.println("Fixed " + p);
                     }
                 } catch (Exception e) {}
             });
        System.out.println("Done");
    }
}
